### PR TITLE
[v17] Refactor `Reauthenticate` components to handle generic MFA challenges.

### PR DESCRIPTION
**Pull request by Joerger**
Backport https://github.com/gravitational/teleport/pull/49680 to branch/v17

e PR - https://github.com/gravitational/teleport.e/pull/5717

Note on /e: This PR is dependent on https://github.com/gravitational/teleport.e/pull/5717 and vice versa. Therefore I'll need to merge this PR with e PR as the head of /e. Then I'll make a follow up to update the e ref to master once the e PR is merged, shortly after this PR is merged.

### CODE REVIEW

File: lib/web/mfa.go [hunk 1]
```diff
@@ -75,6 +75,41 @@ func (h *Handler) deleteMFADeviceWithTokenHandle(w http.ResponseWriter, r *http.
 	return OK(), nil
 }
 
+type deleteMfaDeviceRequest struct {
```
Comment (zmb3) [inline lib/web/mfa.go:78]:
```suggestion
type deleteMFADeviceRequest struct {
```

Acronyms should be capitalized in Go code.

File: lib/web/apiserver.go [hunk 2]
```diff
@@ -920,7 +920,10 @@ func (h *Handler) bindDefaultEndpoints() {
 
 	// MFA private endpoints.
 	h.GET("/webapi/mfa/devices", h.WithAuth(h.getMFADevicesHandle))
+	h.DELETE("/webapi/mfa/devices", h.WithAuth(h.deleteMFADeviceHandle))
 	h.POST("/webapi/mfa/authenticatechallenge", h.WithAuth(h.createAuthenticateChallengeHandle))
+	h.POST("/webapi/mfa/registerchallenge", h.WithAuth(h.createRegisterChallengeHandle))
```
Comment (zmb3) [inline lib/web/apiserver.go:925]: Does the UI code degrade gracefully if it gets a 404 for either of these new endpoints?

Comment (Joerger) [reply lib/web/apiserver.go:925]: They aren't in use yet - 3b46013e069419f867b7361eb4681668d10e2158

File: lib/web/mfa.go (hunk 1 above)
Comment (Joerger) [reply lib/web/mfa.go:78]: 👍 3b0172f378d13cf47a64beed55031add88246ab9

**Comment by Joerger** [comment]
@ryanclark @rosstimothy Friendly ping to review this backport

**Review by ryanclark — APPROVED** [review]

**Comment by Joerger** [comment]
> Looks like the linter isn't happy

Yeah branch/v17 has different lint rules from master for some reason, fixed.

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**